### PR TITLE
Refactor: Use FrontendServiceProxy to call thrift fe rpc

### LINKED ISSUE #4771: Use a generic wrapped client to call frontend thrift rpc
## Refactoring request

**Is your refactoring request related to a problem? Please describe.**

The way to make rpc calls between FEs is too primitive. We need a generic wrapped client to call rpc to avoid having the user manage the conn.

### PR DESCRIPTION
**Pull request by gengjun-git**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4771

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 15 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/master/MasterImpl.java|0|9|00.00%|
|:large_blue_circle:|com/starrocks/external/starrocks/TableMetaSyncer.java|0|6|00.00%|

**Review by Astralidea — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/external/starrocks/TableMetaSyncer.java [hunk 1]
```diff
@@ -24,38 +21,29 @@ public void syncTable(ExternalOlapTable table) {
...
         try {
-            TGetTableMetaResponse response = client.getTableMeta(request);
-            returnToPool = true;
-            List<String> errmsgs = response.status.getError_msgs(); 
+            TGetTableMetaResponse response = FrontendServiceProxy.call(addr, 10000,
+                    client -> client.getTableMeta(request));
             if (response.status.getStatus_code() != TStatusCode.OK) {
-                LOG.info("errmsg: {}", errmsgs.get(0));
+                String errMsg;
+                if (response.status.getError_msgs() != null) {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/external/starrocks/TableMetaSyncer.java:36]: When does the error_msg not been set when StatusCode is not ok?

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/com/starrocks/external/starrocks/TableMetaSyncer.java (hunk 1 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/external/starrocks/TableMetaSyncer.java:36]: This is a protective code.

Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/external/starrocks/TableMetaSyncer.java:36]: We can not assume every one will set the error_msgs.

**Review by chaoyli — APPROVED** [review]

**Pull request merged**